examples/learning_graph_structured_data_h2.py

```diff
@@ -0,0 +1,172 @@
+"""Learning embedding of graph using Poincare Ball Model."""
+
+import logging
+
+import matplotlib.pyplot as plt
+
+import geomstats.backend as gs
+import geomstats.visualization as visualization
+from geomstats.datasets import graph_data_preparation as gdp
+from geomstats.geometry.hyperbolic import Hyperbolic
+
+
+def log_sigmoid(x):
+    """Logsigmoid function."""
+    return gs.log((1 / (1 + gs.exp(-x))))
+
+
+def grad_log_sigmoid(x):
+    """Gradient of logsigmoid function."""
+    return (1 / (1 + gs.exp(x)))
+
+
+def squared_distance(example_embedding, context_embedding,
+                     manifold):
+    """Squared distance considering the manifold metric."""
+    return \
+        manifold.metric.dist(example_embedding, context_embedding)**2
+
+
+def grad_squared_distance(example_embedding, context_embedding,
+                          manifold):
+    """Gradient of the squared distance in the manifold."""
+    log_map =\
+        manifold.metric.log(context_embedding, example_embedding)
+
+    return -2 * log_map
+
+
+def loss(example_embedding, context_embedding, negative_embedding, manifold):
+    """Compute loss and grad.
+
+    Compute loss and grad given embedding of the current example,
+    embedding of the context and negative sampling embedding.
+    """
+    N, D = negative_embedding.shape[0], example_embedding.shape[-1]
+    example_embedding = gs.expand_dims(example_embedding, 0)
+    context_embedding = gs.expand_dims(context_embedding, 0)
+
+    positive_distance =\
+        squared_distance(example_embedding, context_embedding, manifold)
+
+    positive_loss =\
+        log_sigmoid(-positive_distance)
+
+    reshaped_example_embedding =\
+        gs.repeat(example_embedding, N, axis=0)
+
+    negative_distance =\
+        squared_distance(reshaped_example_embedding,
+                         negative_embedding, manifold)
+    negative_loss = log_sigmoid(negative_distance)
+
+    total_loss = -(positive_loss + negative_loss.sum())
+
+    positive_log_sigmoid_grad =\
+        -grad_log_sigmoid(-positive_distance)
+
+    positive_distance_grad =\
+        grad_squared_distance(example_embedding, context_embedding, manifold)
+
+    positive_grad =\
+        gs.repeat(positive_log_sigmoid_grad, D, axis=-1)\
+        * positive_distance_grad
+
+    negative_distance_grad =\
+        grad_squared_distance(reshaped_example_embedding, negative_embedding,
+                              manifold)
+
+    negative_log_sigmoid_grad =\
+        grad_log_sigmoid(negative_distance)
+
+    negative_grad = gs.repeat(negative_log_sigmoid_grad, D, axis=-1)\
+        * negative_distance_grad
+
+    example_grad = -(positive_grad + negative_grad.sum(axis=0))
+
+    return total_loss, example_grad
+
+
+def main():
+    """Learning Poincaré graph embedding.
+
+    Learns Poincaré Ball embedding by using Riemannian
+    gradient descent algorithm
```

Comment:
Missing .

Reply:
Maybe you want to reference one of your papers here? (Using conventions for references in docstrings?)